examples/validation-app/src/interceptors/validate-phone-num.interceptor.ts

```diff
@@ -0,0 +1,91 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: @loopback/example-validation-app
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  bind,
+  Interceptor,
+  InvocationContext,
+  InvocationResult,
+  Provider,
+  ValueOrPromise,
+} from '@loopback/context';
+import {CoffeeShop} from '../models';
+
+/**
+ * This class will be bound to the application as an `Interceptor` during
+ * `boot`
+ */
+@bind({tags: {key: ValidatePhoneNumInterceptor.BINDING_KEY}})
+export class ValidatePhoneNumInterceptor implements Provider<Interceptor> {
+  static readonly BINDING_KEY = `interceptors.${ValidatePhoneNumInterceptor.name}`;
+
+  /*
+  constructor() {}
+  */
+
+  /**
+   * This method is used by LoopBack context to produce an interceptor function
+   * for the binding.
+   *
+   * @returns An interceptor function
+   */
+  value() {
+    return this.intercept.bind(this);
+  }
+
+  /**
+   * The logic to intercept an invocation
+   * It checks the area code of the phone number to make sure it matches
+   * the provided city name.
+   * @param invocationCtx - Invocation context
+   * @param next - A function to invoke next interceptor or the target method
+   */
+  async intercept(
+    invocationCtx: InvocationContext,
+    next: () => ValueOrPromise<InvocationResult>,
+  ) {
+    // Add pre-invocation logic here
+    let coffeeShop: CoffeeShop | undefined;
+    if (invocationCtx.methodName === 'create')
+      Object.assign(coffeeShop, invocationCtx.args[0]);
```

Comment:
Does this work? `coffeeShop` is  undefined at this point. Won't the Object.assign fail if LHS is undefined?

Reply:
you're right. it failed when i'm running the actual app. I've reverted it back to the original:
```
    const coffeeShop: CoffeeShop = new CoffeeShop();
```

If you have a better solution, please let me know!

Reply:
ok. Just had webex with Dom, and we have a solution.  Updated the PR. Thanks!